libs/core/src/lib/tabs/tab-list.component.ts

```diff
@@ -90,106 +152,299 @@ export class TabListComponent implements AfterViewInit, OnChanges, OnDestroy {
-            .subscribe(() => this._resetTabHook());
+    private get _tabPanelsChange$(): Observable<TabPanelComponent[]> {
+        return this.tabPanels.changes.pipe(
+            startWith(this.tabPanels),
+            takeUntil(this._onDestroy$),
+            map(tabPanels => tabPanels.toArray())
+        );
+    }
+
+    /** @hidden */
+    private _detectChanges(): void {
+        if (this._changeDetectorRef && !this._changeDetectorRef['destroyed']) {
+            this._changeDetectorRef.detectChanges();
+        }
+    }
+
+    /** @hidden */
+    private _setupTabPanelsChangeListeners(): void {
+        this._listenOnTabPanelsAndUpdateStorageStructures();
+        this._listenOnTabPanelsAndSetupStackedContent();
+        this._listenOnTabPanelsExpandedChange();
+        this._listenOnTabPanelsAndInitiallyExpandTabPanel();
+    }
+
+    /** @hidden Setup mechanisms required for handling the overflowing tabs behavior */
+    private _setupCollapsingOverflowedTabs(): void {
+        if (this.collapseOverflow) {
+            this._cacheTabsDimensions(this.tabHeaders.toArray());
+            this._listenOnTabPanelsChangeAndCollapse();
+            this._listenOnResizeAndHideItems();
+            this._hideOverflowingItems();
+            this._keepActiveTabVisible();
+        } else {
+            this._isCollapsed = false;
+        }
+    }
+
+    /** @hidden Setup mechanisms required for handling the stacked content behavior */
+    private _listenOnTabPanelsAndSetupStackedContent(): void {
+        if (this.stackContent) {
+            this._tabPanelsChange$.pipe(delay(0))
+                .subscribe(() => this._tabArray
+                    .filter(tab => !tab.panel.disabled)
+                    .forEach(tab => tab.panel._expand(true)
+                    )
+                );
+        }
+    }
+
+    /** @hidden */
+    private _listenOnTabPanelsAndUpdateStorageStructures(): void {
+        this._tabPanelsChange$.pipe(
+            map(tabPanels => tabPanels.map(el => new TabInfo(el)))
+        ).subscribe(tabs => {
+            this._tabArray = tabs;
+            this._numbOfVisibleTabs = tabs.length;
+            this._resetVisualOrder();
+        });
+    }
+
+    /** @hidden */
+    private _listenOnTabPanelsExpandedChange(): void {
+        this._tabPanelsChange$.pipe(
+            map(tabPanels => tabPanels.map(el => el._expandedStateChange.asObservable())),
+            switchMap(tabPanels => merge(...tabPanels))
+        ).subscribe(event => this._expandTab(event.target, event.state));
+    }
+
+    /** @hidden */
+    private _listenOnTabPanelsAndInitiallyExpandTabPanel(): void {
+        this._tabPanelsChange$.pipe(
+            map(_ => this._tabArray.find(tab => tab.active && !tab.disabled)),
+            filter(tab => !tab),
+            map(_ => this._tabArray.find(tab => !tab.disabled)),
+            filter(tab => !!tab),
+            delay(0)
+        ).subscribe(tab => this._expandTab(tab.panel, true));
     }
```

Comment:
How is it going to work? There are 2 filters, that are excluding eachother

Reply:
This is working well:
- first filter check whether there is a active tab
- second filter check whether there is a tab that might be activated

Actually this can be simplified to:
```
        this._tabPanelsChange$.pipe(
            filter(_ => !this._tabArray.some(tab => tab.active)),
            map(_ => this._tabArray.find(tab => !tab.disabled)),
            filter(Boolean),
            delay(0)
        ).subscribe(tab => this._expandTab(tab.panel, true));
```